src/api/routes/block.ts

```diff
@@ -18,92 +20,119 @@ const parseBlockQueryLimit = parseLimitQuery({
 
 export function createBlockRouter(db: DataStore): RouterWithAsync {
   const router = addAsync(express.Router());
+  const cacheHandler = getChainTipCacheHandler(db);
+  router.get(
+    '/',
+    cacheHandler,
+    asyncHandler(async (req, res) => {
```

Comment:
Note for reviewers: we've been using `router.getAsync`, an Express extension provided by the [`@awaitjs/express` lib](https://www.npmjs.com/package/@awaitjs/express). However, the `getAsync` function has incorrect/bugged behavior when given multiple router handler functions. It executes both synchronously at the same time. This breaks the ability for the handlers registered first to control the route flow -- in the case of this PR, prevents caching handler from being able to skip the non-caching handler.

In addition, the `@awaitjs/express` lib is incorrectly typed and is forcing this repo to disable linting for `no-misused-promises`. The author of that lib has rejected PRs attempting to fix the typing.